postgresql/src/main/java/org/apache/zeppelin/postgresql/SqlCompleter.java

```diff
@@ -0,0 +1,200 @@
+    String driverSpecificKeywords =
+        "/" + metaData.getDriverName().replace(" ", "-").toLowerCase() + "-sql.keywords";
+
+    logger.info("JDBC DriverName:" + driverSpecificKeywords);
+
+    if (SqlCompleter.class.getResource(driverSpecificKeywords) != null) {
+      logger.info("JDBC DriverName: 2" + driverSpecificKeywords);
+      String driverKeywords =
+          new BufferedReader(new InputStreamReader(
+              SqlCompleter.class.getResourceAsStream(driverSpecificKeywords))).readLine();
+      keywords += "," + driverKeywords.toUpperCase();
+    }
+
+    Set<String> completions = new TreeSet<String>();
+
+
+    // Add the keywords from the current JDBC connection
+    try {
+      keywords += "," + metaData.getSQLKeywords();
+    } catch (Exception e) {
+      logger.debug("fail to get SQL key words from database metadata: " + e, e);
+    }
+    try {
+      keywords += "," + metaData.getStringFunctions();
+    } catch (Exception e) {
+      logger.debug("fail to get string function names from database metadata: " + e, e);
+    }
+    try {
+      keywords += "," + metaData.getNumericFunctions();
+    } catch (Exception e) {
+      logger.debug("fail to get numeric function names from database metadata: " + e, e);
+    }
+    try {
+      keywords += "," + metaData.getSystemFunctions();
+    } catch (Exception e) {
+      logger.debug("fail to get system function names from database metadata: " + e, e);
+    }
+    try {
+      keywords += "," + metaData.getTimeDateFunctions();
+    } catch (Exception e) {
+      logger.debug("fail to get time date function names from database metadata: " + e, e);
+    }
+
+    // Also allow lower-case versions of all the keywords
+    keywords += "," + keywords.toLowerCase();
+
+    StringTokenizer tok = new StringTokenizer(keywords, ", ");
+    while (tok.hasMoreTokens()) {
+      completions.add(tok.nextToken());
+    }
+
+    // now add the tables and columns from the current connection
+    if (!(skipmeta)) {
+      String[] columns = getColumnNames(connection.getMetaData());
+      for (int i = 0; columns != null && i < columns.length; i++) {
+        completions.add(columns[i++]);
+      }
+    }
+
+    return completions;
+  }
+
+  private static String[] getColumnNames(DatabaseMetaData meta) throws SQLException {
+    Set<String> names = new HashSet<String>();
+
+    try {
+      ResultSet columns = getColumns(meta, "%");
+      try {
+
+        while (columns.next()) {
+          // add the following strings:
+          // 1. column name
+          // 2. table name
+          // 3. tablename.columnname
+          String name = columns.getString("TABLE_NAME");
+          names.add(name);
+          names.add(columns.getString("COLUMN_NAME"));
+          names.add(columns.getString("TABLE_NAME") + "." + columns.getString("COLUMN_NAME"));
+        }
+      } finally {
+        columns.close();
+      }
+
+      return names.toArray(new String[0]);
+    } catch (Throwable t) {
```

Comment:
nit: the indentation is a bit weird in this method. is it possible to align the catch with the try?